frictionless_ckan_mapper/ckan_to_frictionless.py

```diff
@@ -1,118 +1,123 @@
 
         # TODO: remove as CKAN Resource does not have extras (CKAN has already
-        # # unpacked them)
+        # unpacked them)
         if 'extras' in resource:
             resource['extras'] = json.loads(resource['extras'])
 
-        # unjsonify values
-        # 1. check if string
-        # 2. if starts with [ or { => json loads it ...
-            # slightly hacky way to check if value is a jsonified array or dict
-        # 3. else do nothing 
-        for k,v in resource.items():
-            if isinstance(v, str):
-                v = v.strip()
-                if v.startswith('{') or v.startswith('['):
+        # Unjsonify values
+        # slightly hacky way to check if value is a jsonified array or dict
+        for key, value in resource.items():
+            if isinstance(value, str):
+                value = value.strip()
+                if value.startswith('{') or value.startswith('['):
                     try:
-                        v = json.loads(v)
-                        resource[k] = v
-                    except (json.decoder.JSONDecodeError, TypeError):
+                        value = json.loads(value)
+                        resource[key] = value
+                    except (json_parse_exception, TypeError):
                         pass
 
         # Reformat expected output for some keys in resource
         # resource['format'] = resource['format'].lower()
         if 'name' in resource:
             resource['name'] = slugify.slugify(resource['name']).lower()
 
-        # Remap differences from CKAN to Frictionless resource
-        for k, v in self.resource_mapping.items():
-            if k in resource:
-                resource[v] = resource[k]
-                del resource[k]
-
-                # Cast CKAN resource size to int
-                if k == 'size':
-                    if not resource[v]:  # receives `null`
-                        resource[v] = 0
-                    else:
-                        resource[v] = int(resource[v])
+        resource = self._remap_resource_ckan_to_frictionless(resource)
         return resource
 
-    package_mapping = {
-        'notes': 'description',
-        'tags': 'keywords',  # this is flattened and simplified
-    }
-
-    package_sources_mapping = {
-        'author': 'title',
-        'author_email': 'email',
-        'url': 'path',
-    }
-
-    def dataset(self, ckandict):
-        '''Convert a CKAN Package (Dataset) to Frictionless Package.
-
-        1. Remove unneeded keys
-        2. Expand extras.
-            * ~~Apply heuristic to unjsonify (if starts with [ or { unjsonify~~
-            * JSON loads everything and on error have a string
-        3. Map keys from CKAN to Frictionless (and reformat if needed)
-        4. Remove keys with null values (CKAN has a lot of null valued keys)
-        4. Apply special formatting for key fields
-        '''
-        outdict = dict(ckandict)
+    @staticmethod
```

Comment:
Refactoring like this _may_ look a bit overkill, but this is from `pylama` evaluating with `mccabe` the complexity of functions and there is no way around unless we may want to tweak the settings to allow more complex functions.

https://www.flake8rules.com/rules/C901.html